File: Backend/PicturesAPI/app/models/picture.py
from sqlalchemy import Column, Integer, BIGINT, String, DateTime, func, text
from app.database.db import Base


class Picture(Base):
    __tablename__ = "pictures"
    id = Column(BIGINT, primary_key=True, index=True)
    file_name = Column(String(255), unique=True, index=True)
    file_path = Column(String(255))
    file_size = Column(Integer)
    height = Column(Integer)
    width = Column(Integer)
    date_created = Column(DateTime(timezone=True), server_default=func.now())
    date_updated = Column(DateTime(timezone=True), server_default=text("NULL"), onupdate=func.now())

    def __repr__(self):
        return 'Picture entry:' + \
               f'id: {self.id}\n' + \
               f'file_name: {self.file_name}\n' + \
               f'file_path: {self.file_path}\n' + \
               f'file_size: {self.file_size}\n' + \
               f'height: {self.height}\n' + \
               f'width: {self.width}\n' + \
               f'date_created: {self.date_created}\n' + \
               f'date_updated: {self.date_updated}\n'

File: app/lib/models/picture.dart
class Picture {
  final int id;
  final String fileName;
  final String filePath;
  final int fileSize;
  final int width;
  final int height;
  final String dateCreated;
  final String dateUpdated;

  Picture(this.id, this.fileName, this.filePath, this.fileSize, this.width,
      this.height, this.dateCreated, this.dateUpdated);

  Picture.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        fileName = json['file_name'],
        filePath = json['file_path'],
        fileSize = json['file_size'],
        width = json['width'],
        height = json['height'],
        dateCreated = json['date_created'],
        dateUpdated = json['date_updated'];

  Map<String, dynamic> toJson() => {
        'id': this.id,
        'file_name': this.fileName,
        'file_path': this.filePath,
        'file_size': this.fileSize,
        'width': this.width,
        'height': this.height,
        'date_created': this.dateCreated,
        'date_updated': this.dateUpdated
      };

  @override
  String toString() {
    return '''Picture with ID = ${this.id}:
    Filename: ${this.fileName}
    File path: ${this.filePath}
    File size: ${this.fileSize}
    Width: ${this.width}
    Height: ${this.height}
    Date created: ${this.dateCreated}
    Date updated: ${this.dateUpdated}
    ''';
  }
}
